### DISCUSSION TITLE
Separation of Function Type Declaration and Definition

### DISCUSSION THREAD

**Discussion by pharringtonp19** [google-research/dex-lang#942] [Q&A] [answered]
Is it possible to separate the type declaration of a function from the definition as done in Haskell?

```
f :: Float32 -> Float32
f x = x*x 
```

**Comment by dougalm** [comment, marked as answer]
No, you have to write either `f : Float32 -> Float32 = \x. x * x` or, more conventionally, use the `def` syntax, `def (x:Float32) : Float32 = x * x`. The latter is just syntactic sugar for the former. I never much liked the Haskell style because you have to write the function's name twice.

**Reply by pharringtonp19** [reply]
@dougalm Thanks for the explanation
